helix-term/src/commands.rs

```diff
@@ -4234,3 +4236,30 @@ fn suspend(_cx: &mut Context) {
     #[cfg(not(windows))]
     signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
 }
+
+fn add_newline_above(cx: &mut Context) {
+    add_newline_impl(cx, true);
+}
+
+fn add_newline_below(cx: &mut Context) {
+    add_newline_impl(cx, false)
+}
+
+fn add_newline_impl(cx: &mut Context, above: bool) {
+    let count = cx.count();
+    let (view, doc) = current!(cx.editor);
+    let selection = doc.selection(view.id);
+    let text = doc.text();
+    let slice = text.slice(..);
+
+    let changes = selection.into_iter().map(|range| {
+        let (start, end) = range.line_range(slice);
+        let line = if above { start } else { end + 1 };
+        let pos = text.line_to_char(line);
+        (pos, pos, Some("\n".repeat(count).into()))
```

Comment:
Helix supports multiple possible line endings, so you shouldn't hard-code `\n` here.  Use the document's line ending (`doc.line_ending.as_str()`) instead.